driver/main.cpp

```diff
@@ -7,6 +7,20 @@
 // License, use of this software will be governed by the Apache License, Version 2.0,
 // included in the file licenses/APL.txt.
 
+#ifdef _WIN32
+#ifndef WIN32_LEAN_AND_MEAN
+#define WIN32_LEAN_AND_MEAN
+#endif
+
+#ifndef NOMINMAX
+#define NOMINMAX
+#endif
+
+#include <winsock2.h>
```

Comment:
What are these for?

Reply:
just copying what i said in slack:

 the new stuff in main.cpp is because once i added a device manager to the other files, it for some reason links things in a way that causes an error because of symbols included in both windows and winsock2.h . its a known issue and the accepted solution is to add those ifdefs and explicitly define those libraries in that exact order
 
 
![image](https://github.com/user-attachments/assets/dd6c6081-d60b-4a6a-9cd5-6508b26e4a54)